src/lib_json/json_value.cpp

```diff
@@ -529,7 +529,7 @@ bool Value::operator<(const Value& other) const {
   }
   case arrayValue:
   case objectValue: {
-    int delta = int(value_.map_->size() - other.value_.map_->size());
+    auto delta = int(value_.map_->size() - other.value_.map_->size());
```

Comment:
This looks like a dangerous downcast of a potentially "negative" `size_t` to `int`, which is implementation-defined behavior. We should make the relatively easy change to make the calculation portable. We should also lose the training wheels on the following line's direct map comparison.

```
    auto thisSize = value_.map_->size();
    auto otherSize = other.value_.map_->size();
    if (thisSize != otherSize)
      return thisSize < otherSize;
    *value_.map_ < *other.value_.map_;
```